### ISSUE TITLE
TypeError when passing custom eventEmitter to createResponse

### ISSUE THREAD

**Issue by joerx** [howardabrams/node-mocks-http#62]
Caused by below piece of code (in `mockResponse.js`). `eventEmitter` passed in via options overrides included `EventEmitter` module, then trying to create new instance triggers `TypeError: object is not a function`

```js
    var EventEmitter = require('./mockEventEmitter');
    //...
    if (options.eventEmitter) {
        EventEmitter = options.eventEmitter;
    }
    var writableStream = new WritableStream();
    var eventEmitter = new EventEmitter();
```

**Comment by JohnnyEstilles** [comment]
@joerx, thanks for reporting this. I have not been able to reproduce the issue. If I understood you correctly, the error is thrown when you pass a custom event emitter to `mockRepsonse`. I my interpretation correct?

Would you be kind enough to show your code, to help me reproduce the error on my end?

Thanks!

**Comment by joerx** [comment]
It was actually a misconception on my side on how to use your api. However, I feel the 'correct' way it rather counterintuitive. I created a gist to illustrate my problem: https://gist.github.com/joerx/6b387bb3dfe00d36573f

It boils down to this piece:
```js
var res = httpMocks.createResponse({
  eventEmitter: EventEmitter            // works, but is somewhat counterintuitive
  // eventEmitter: new EventEmitter()   // this is what I would expect to work
});
```

The issue is that `createResponse` expects the constructor for EventEmitter to be passed in, not an actual emitter, as I expected. It would be good to have at least a sample in the documentation to illustrate the correct usage.

**Comment by joerx** [comment]
To be more precise, I would expect to have the event-system in a mocked response to actually work out of the box, so asynchronous code can be tested by listening to the `end` event. With the current default (using a bogus emitter) only synchronous handler code could be tested. Or am I missing something?

**Comment by JohnnyEstilles** [comment]
@joerx, unfortunately, the event system in the current release does not work out of the box. As you already figured out, it requires you to BYO event emitter (so to speak). That said, we do intend have `mockResponse` inherit from Node's `EventEmitter` in the next version. Feel free to check out our `2.0` roadmap (issue #54). Comments are always welcome.

**Comment by JohnnyEstilles** [comment]
I have also updated our [README](https://github.com/howardabrams/node-mocks-http) with a note explaining this situation. Thanks for bringing up the issue.

**Comment by JohnnyEstilles** [comment]
I will consider this issue resolved (docs updated via f923e144eafc5d4a134377832816ba1b31eebc63). I'm closing the issue, but please feel free to re-open it if you need to.
